motor/providers/azure/authentication.go

```diff
@@ -0,0 +1,52 @@
+package azure
+
+import (
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/cockroachdb/errors"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery/motor/vault"
+)
+
+// GetAuthorizer determines what authorizer to use, based on the passed in configs. Possible options are:
+// - Authorizer via the CLI that uses the `az` executable
+// - Authorizer via password that uses the sdk
+// - Authorizer via certificate that uses the sdk
+func GetAuthorizer(clientId, tenantId, resource string, credential *vault.Credential) (autorest.Authorizer, error) {
```

Comment:
Moved this here from `provider.go` to let us use it w/o neeeding to actually have a provider (i.e. when discovering intitially).